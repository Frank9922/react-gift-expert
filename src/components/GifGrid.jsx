import { getGif } from "../helpers/getGifs";
import { useState, useEffect } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ( { category, RemoveCategory } ) => {

    const { imagenes, isLoading } = useFetchGifs(category);

    return (
        <>
            <div>
            <div className="card-title">
            <h3>{category}</h3>
            {
                isLoading ? (<h2>Cargando...</h2>) : null
            }
            <button onClick={() => {RemoveCategory(category)}} className="btn-eliminar">Eliminar</button>
            </div>

            <div className="card-grid">
                {
                    imagenes.map( (image) => (
                        <GifItem key={image.id}
                        {...image}
                        
                        />
                    ))
                }
            </div>
            </div>

            
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}