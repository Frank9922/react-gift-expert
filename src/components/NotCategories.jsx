
export const NotCategories = () => {
  return (
    <div className="not-categories">
        <h1>No hay categorias ...</h1>
    </div>
  )
}
