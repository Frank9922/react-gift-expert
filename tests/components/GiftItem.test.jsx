import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en el componente <GiftItem />', () => {

    const title = "Test FLeiva"
    const url = 'https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=fd687d8auic34croy17n9dqk9o4uzmb8k6ntrvmh9v0w57h0&ep=v1_gifs_search&rid=giphy.gif&ct=g'


    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar el URL y el ALT indicado', () => {

        const { container } = render(<GifItem title={title} url={url} />);

        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe de mostrar el titulo en el componente', () => {

        const { container } = render(<GifItem title={title} url={url} />);


        expect(screen.getByText(title)).toBeTruthy();
    })
})