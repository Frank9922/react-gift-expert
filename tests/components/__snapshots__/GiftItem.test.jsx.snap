// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GiftItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test FLeiva"
      src="https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=fd687d8auic34croy17n9dqk9o4uzmb8k6ntrvmh9v0w57h0&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Test FLeiva
    </p>
  </div>
</div>
`;
